### PR TITLE
UTY-2714: Incorrect entity added/removed callback registration

### PR DESCRIPTION
**Pull request by BryanJY-Wong**
#### Description
Fix an incorrect callback registration

#### Tests
This bug doesn't seem to be causing any problems as entity removal is processed by the `GameObjectInitializationSystem`. But just to be safe, I did a local launch, spawned some cubes and shot them off the platform.

#### Documentation
- [x] Changelog

#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: workers/unity/Packages/packages-lock.json [hunk 1]
```diff
@@ -0,0 +1,389 @@
+{
```
Comment (jamiebrynes7) [inline workers/unity/Packages/packages-lock.json:1]: This file is explicitly ignored by the [`gitignore`](https://github.com/spatialos/gdk-for-unity/blob/develop/workers/unity/.gitignore#L10).

How did you manage to check this in? In any case, this shouldn't be here.

**Review by jamiebrynes7 — CHANGES_REQUESTED** [review]

File: workers/unity/Packages/packages-lock.json (hunk 1 above)
Comment (BryanJY-Wong) [reply workers/unity/Packages/packages-lock.json]: I think I didn't pull the develop branch before I made this branch 🤦

Comment (jamiebrynes7) [reply workers/unity/Packages/packages-lock.json]: Ahhhh, that would explain it 😂

File: CHANGELOG.md [hunk 2]
```diff
@@ -31,6 +31,10 @@
 - Fixed an issue where authority changes returned by `ComponentUpdateSystem.GetAuthorityChangesReceived()` were returned in order from newest to oldest. [#1465](https://github.com/spatialos/gdk-for-unity/pull/1465)
 - Fixed a bug where the build system would throw a null reference exception if you don't have a configuration for a worker type. [#1461](https://github.com/spatialos/gdk-for-unity/pull/1461)
 
+### Fixed
+
+- Fixed an incorrect callback registration when entities are created/removed request is sent by the `WorldCommandSender`. [#1473](https://github.com/spatialos/gdk-for-unity/pull/1462)
```
Comment (jamiebrynes7) [inline CHANGELOG.md:36]: There's already a fixed section right above this 😛

**Review by jamiebrynes7 — APPROVED** [review]

**Pull request merged**